web/modules/custom/joinup_core/src/EventSubscriber/JoinupCoreOgSubscriber.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\joinup_core\EventSubscriber;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\og\Event\PermissionEventInterface as OgPermissionEventInterface;
+use Drupal\og\GroupPermission;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Joinup core subscriber for og related events.
+ */
+class JoinupCoreOgSubscriber implements EventSubscriberInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      OgPermissionEventInterface::EVENT_NAME => [['provideOgGroupPermissions']],
+    ];
+  }
+
+  /**
+   * Declare OG permissions for shared entities.
+   *
+   * @param \Drupal\og\Event\PermissionEventInterface $event
+   *   The OG permission event.
+   */
+  public function provideOgGroupPermissions(OgPermissionEventInterface $event) {
+    $event->setPermission(
+      new GroupPermission([
+        'name' => 'administer shared entities',
+        'title' => $this->t('Administer shared entities'),
+        'restrict access' => TRUE,
+      ])
+    );
+  }
```

Comment:
This entire file seems to be unneeded and duplicating the code in `\Drupal\joinup_community_content/EventSubscriber/EventSubscriber`. I tried to comment out this code and the test still completes without problems. This seems to make sense since this subscriber is not defined in `joinup_core.services.yml` so it is never called.

Reply:
The `administer shared entities` was removed from the `joinup_community_content`'s event subscriber as now, solutions can also be shared, thus the permission belongs to a higher level module like `joinup_core`. Leaving the necessity to our private chat and will report back here.